### PR TITLE
[core] Use `Status::AlreadyExists` instead of `Status::NotFound` when a named actor already exists

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

It is weird to return `NotFound` when a named actor already exists when creating an actor.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/common/status.cc [hunk 1]
```diff
@@ -53,6 +53,7 @@ namespace ray {
 #define STATUS_CODE_NOT_FOUND "NotFound"
 #define STATUS_CODE_DISCONNECTED "Disconnected"
 #define STATUS_CODE_SCHEDULING_CANCELLED "SchedulingCancelled"
+#define STATUS_CODE_ALREADY_EXISTS "AlreadyExists"
```
Comment (dentiny) [inline src/ray/common/status.cc:56]: (no action needed, not high priority, no need to be in this PR) Is there any chance we use string?
Macros for constants are discouraged.
Ref: effective C++ item-2.

**Review by dentiny — APPROVED** [review]
Nice!

**Comment by jjyao** [comment]
Test failures

File: src/ray/common/status.cc (hunk 1 above)
Comment (rynewang) [reply src/ray/common/status.cc:56]: No objections. We can remove all macros and just put string literals into kCodeToStr, other than the STATUS_CODE_UNKNOWN which we can make a constexpr string_view.

**Pull request merged**